nsxt/resource_nsxt_policy_lb_http_monitor_profile.go

```diff
@@ -0,0 +1,207 @@
+	tags := getPolicyTagsFromSchema(d)
+	requestBody := d.Get("request_body").(string)
+	requestHeaders := getPolicyLbHTTPHeaderFromSchema(d, "request_header")
+	requestMethod := d.Get("request_method").(string)
+	requestURL := d.Get("request_url").(string)
+	requestVersion := d.Get("request_version").(string)
+	responseBody := d.Get("response_body").(string)
+	responseStatusCodes := interface2Int64List(d.Get("response_status_codes").([]interface{}))
+	fallCount := int64(d.Get("fall_count").(int))
+	interval := int64(d.Get("interval").(int))
+	riseCount := int64(d.Get("rise_count").(int))
+	timeout := int64(d.Get("timeout").(int))
+	monitorPort := int64(d.Get("monitor_port").(int))
+
+	resourceType := model.LBMonitorProfile_RESOURCE_TYPE_LBHTTPMONITORPROFILE
+
+	obj := model.LBHttpMonitorProfile{
+		DisplayName:         &displayName,
+		Description:         &description,
+		Tags:                tags,
+		RequestBody:         &requestBody,
+		RequestHeaders:      requestHeaders,
+		RequestMethod:       &requestMethod,
+		RequestUrl:          &requestURL,
+		RequestVersion:      &requestVersion,
+		ResponseBody:        &responseBody,
+		ResponseStatusCodes: responseStatusCodes,
+		FallCount:           &fallCount,
+		Interval:            &interval,
+		MonitorPort:         &monitorPort,
+		RiseCount:           &riseCount,
+		Timeout:             &timeout,
+		ResourceType:        resourceType,
+	}
+
+	log.Printf("[INFO] Patching LBHttpMonitorProfile with ID %s", id)
+	dataValue, errs := converter.ConvertToVapi(obj, model.LBHttpMonitorProfileBindingType())
+	if errs != nil {
+		return fmt.Errorf("Error converting LBHttpMonitorProfile %s", errs[0])
+	}
+
+	client := infra.NewLbMonitorProfilesClient(connector)
+	return client.Patch(id, dataValue.(*data.StructValue))
+}
+
+func resourceNsxtPolicyLBHttpMonitorProfileCreate(d *schema.ResourceData, m interface{}) error {
+
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID(d, m, resourceNsxtPolicyLBHttpMonitorProfileExists)
+	if err != nil {
+		return err
+	}
+
+	err = resourceNsxtPolicyLBHttpMonitorProfilePatch(d, m, id)
+	if err != nil {
+		return handleCreateError("LBHttpMonitorProfile", id, err)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+
+	return resourceNsxtPolicyLBHttpMonitorProfileRead(d, m)
+}
+
+func resourceNsxtPolicyLBHttpMonitorProfileRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	converter := bindings.NewTypeConverter()
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining LBHttpMonitorProfile ID")
+	}
+
+	client := infra.NewLbMonitorProfilesClient(connector)
+	obj, err := client.Get(id)
+	if err != nil {
+		return handleReadError(d, "LBHttpMonitorProfile", id, err)
+	}
+
+	baseObj, errs := converter.ConvertToGolang(obj, model.LBHttpMonitorProfileBindingType())
+	if len(errs) > 0 {
+		return fmt.Errorf("Error converting LBHttpMonitorProfile %s", errs[0])
```

Comment:
I think this error that would appear if we try to import monitor of the wrong type. Lets make it more verbose, something like "lb monitor with id xxx is not of http type"